Rover/fence.cpp

```diff
@@ -18,32 +18,34 @@ void Rover::fence_check()
     // if there is a new breach take action
     if (new_breaches) {
         // if the user wants some kind of response and motors are armed
-        if (fence.get_action() != Failsafe_Action_None) {
+        if ((FailsafeAction)fence.get_action() != FailsafeAction::None) {
             // if within 100m of the fence, it will take the action specified by the FENCE_ACTION parameter
             if (fence.get_breach_distance(new_breaches) <= AC_FENCE_GIVE_UP_DISTANCE) {
-                switch (fence.get_action()) {
-                case Failsafe_Action_None:
+                switch ((FailsafeAction)fence.get_action()) {
+                case FailsafeAction::None:
                     break;
-                case Failsafe_Action_RTL:
-                    if (!set_mode(mode_rtl, ModeReason::FENCE_BREACHED)) {
-                        set_mode(mode_hold, ModeReason::FENCE_BREACHED);
+                case FailsafeAction::SmartRTL:
+                    if (set_mode(mode_smartrtl, ModeReason::BATTERY_FAILSAFE)) {
+                        break;
                     }
-                    break;
-                case Failsafe_Action_Hold:
-                    set_mode(mode_hold, ModeReason::FENCE_BREACHED);
-                    break;
-                case Failsafe_Action_SmartRTL:
-                    if (!set_mode(mode_smartrtl, ModeReason::FENCE_BREACHED)) {
-                        if (!set_mode(mode_rtl, ModeReason::FENCE_BREACHED)) {
-                            set_mode(mode_hold, ModeReason::FENCE_BREACHED);
-                        }
+                    FALLTHROUGH;
+                case FailsafeAction::RTL:
+                    if (set_mode(mode_rtl, ModeReason::BATTERY_FAILSAFE)) {
+                        break;
                     }
+                    FALLTHROUGH;
+                case FailsafeAction::Hold:
+                    set_mode(mode_hold, ModeReason::BATTERY_FAILSAFE);
                     break;
-                case Failsafe_Action_SmartRTL_Hold:
+                case FailsafeAction::SmartRTL_Hold:
                     if (!set_mode(mode_smartrtl, ModeReason::FENCE_BREACHED)) {
                         set_mode(mode_hold, ModeReason::FENCE_BREACHED);
                     }
                     break;
+                case FailsafeAction::Terminate:
+                    set_mode(mode_hold, ModeReason::FAILSAFE);
```

Comment:
I'm not sure we need to set the mode .. simply disarming is closer to what "terminate" means I think.

Reply:
Agree. I am not sure either.
My reasoning was the following : Terminate want the drone to stop and not move anymore. Disarm can work. 
But : Rover/boat have auto-arm (slope, river), CC can relaunch the arming and we will have endless issue, anything that will rearm will retrigger what cause the termination as the rover/boat will continue what it was doing when in non manual control. 
For those reasons, I add the hold so any rearm won't move the drone into a faulty/dangerous position, unless the operator explicitly ask for it.

Reply:
Re the possible interference with a companion computer, I don't think we should think too far ahead.  Let's just do the minimum and keep it simple.

Reply:
ok